back/app/services/participants_service.rb

```diff
@@ -174,4 +174,40 @@ def idea_statuses_participants(idea_statuses, options = {})
     ideas = Idea.where(idea_status: idea_statuses)
     ideas_participants ideas, options
   end
+
+  def reset_participation_data(project)
+    # Reset volunteers data
+    Volunteering::Volunteer
+      .joins(cause: { phase: :project })
+      .where(cause: { phases: { project_id: project.id } })
+      .destroy_all
+
+    # Reset event attendance data
+    Events::Attendance
+      .joins(event: :project)
+      .where(events: { project_id: project.id })
+      .destroy_all
+
+    # Reset poll data
+    Polls::Response
+      .joins(phase: :project)
+      .where(phases: { project_id: project.id })
+      .destroy_all
+
+    # Reset basket data
+    Basket.where(phase: Phase.where(project: project)).destroy_all
+
+    # Filter ideas that belong to a voting phase, and destroy all comments and reactions that they have.
+    ideas = Idea.where(project: project).where(
+      id: IdeasPhase.where(phase: Phase.where(project: project, participation_method: 'voting')).select(:idea_id)
+    )
+
+    ideas.each do |idea|
+      idea.comments.destroy_all
+      idea.reactions.destroy_all
+    end
```

Comment:
```suggestion
    voting_phases = Phase.where(project: project, participation_method: 'voting')
    ideas = Idea.joins(:phases).where(phases: voting_phases)
    Comment.where(post: ideas).destroy_all
    Reaction.where(reactable: ideas).destroy_all
```
triggers less queries.

Yet another, and maybe simpler solution, would be to delete all reactions and comments of the project:
```ruby
    Comment.where(post: project.ideas).destroy_all
    Reaction.where(reactable: project.ideas).destroy_all
```

Reply:
On a higher level, I think this is a good heuristics, but it might be challenging to explain to customers and clear their doubts about what will be deleted and what will not. Also, it's not covering the "option analysis" use case. Should we extend this approach to ideation phases where posting is not allowed? (which will make it even more difficult to explain 😅)

Reply:
I think we should probably leave it as it is for right now and see whether we need to extend it further based on the usage. Option analysis is not technically a participation method + there might be other reasons posting is now allowed so I feel this is probably going a step too far. l I would ask Rob in the channel about what he thinks about the current copy though, as there might be a need to clarify the voting behaviour a bit.